pluginlib/pluginlib-extras.cmake

```diff
@@ -35,3 +35,7 @@ if(UNIX AND NOT APPLE)
   # ament_export_libraries() because it is not absolute and cannot be found with find_library
   list(APPEND pluginlib_LIBRARIES ${FILESYSTEM_LIB})
 endif()
+
+find_package(tinyxml2_vendor REQUIRED)
```

Comment:
I'm somewhat hesitant about finding the vendor package here as someday we won't need one for tinyxml2 and tinyxml2_vendor when used at install time should register CMake configuration that locates TinyXML2 for builds using the same install space... but maybe that doesn't work for isolated builds when tinyxml2_vendor is in the same workspace.

It's probably fine as-is but I'm curious if removing tinyxml2_vendor still allows successful cross-compilation with tinyxml2_vendor in an underlay and in the same workspace as pluginlib.

Reply:
I recall testing these changes without `tinyxml2_vendor` and not working, but it's been a while.

Reply:
I think it makes sense for it not to work if tinyxml2_vendor is in the same workspace but if TinyXML2 is available from the base system or in an underlay workspace via tinyxml2_vendor it should work without it. But that's not a condition worth testing for and we'll have to pull the vendor packages out by hand if we retire them no matter what.